### PR TITLE
Add lookml-gen

### PR DESCRIPTION
**Pull request by akhatre**
Adding `lookml-gen` command to generate LookML view files like [this one](https://mondobank.looker.com/projects/mondobank/files/risky_user_connections.view.lkml). We already have that in [dbt monzo](https://monzo.slack.com/archives/C75MDGSES/p1599833518129400), but as with other commands ddbt is much faster.

The commands runs the following steps:
* Compile all the models
* Find the filesystem model and target based on the execution context
* Get the columns and data types from the model in BigQuery (using the target to focus on the dev branch for the user)
* Generate string of LookML file structure, mapping data types from BigQuery to Looker and adding field specific blocks
* Copy the generated string to clipboard for user to paste into their Looker online interface

The command does not support `-m` flag for multiple models selection. I am not sure this is a desired functionality since usually its is just one lookml file that needs generation and the contents of that file are pasted into browser rather than being saved into a repository.

### CODE REVIEW

File: cmd/lookml_gen.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// itereate over fields, record field names and data types
+	for _, fieldSchema := range schema {
+		column := fmt.Sprintf("%v", fieldSchema.Name)
+		dtype := fmt.Sprintf("%v", fieldSchema.Type)
+		columns = append(columns, column)
+		dtypes = append(dtypes, dtype)
```
[the remark this answers is not in the document]
Comment (akhatre) [reply cmd/lookml_gen.go]: Agree, now adding `fieldSchema.Name` directly. `fieldSchema.Type` is of custom type derived from string, so this one still needs conversion I think.

File: cmd/lookml_gen.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/atotto/clipboard"
+)
+
+// map bigquery data types to looker data types
+var mapBqToLookerDtypes map[string]string = map[string]string{
```
[the remark this answers is not in the document]
Comment (akhatre) [reply cmd/lookml_gen.go:16]: Mmm, what was strange? Ran `gofmt` on the file and it aligned keys and values vertically using varying number of spaces, does this make alignment as expected?

File: cmd/lookml_gen.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	}
+
+	// initialise lookml view head
+	lookmlView := "view: " + modelName +" {\n\n"
+	lookmlView += "sql_table_name: `" + target.ProjectID + "." + target.DataSet + "." + modelName + "` ;;\n"
+
+	// add dimensions and appropriate blocks for each field
+	for i := 0; i < len(bqColumns); i ++ {
+		colName := bqColumns[i]
+		colDtype := mapBqToLookerDtypes[bqDtypes[i]]
```
[the remark this answers is not in the document]
Comment (akhatre) [reply cmd/lookml_gen.go:106]: Good spot, added check, will return error if colDtype == "".

**Comment by akhatre** [comment]
@nickng @DomBlack please let me know if there are any other changes needed 👀

**Review by DomBlack — APPROVED** [review]
Looks good to me, was going to approve after lunch :-)

**Pull request merged**